app/utils/recirculationBlacklist.js

```diff
@@ -0,0 +1,53 @@
+import localStorageConnector from '@wikia/ember-fandom/utils/local-storage-connector';
+
+const CACHE_KEY = 'recommendations_blacklist';
+const CURRENT_SESSION_KEY = (window && window.Cookies) ? window.Cookies.get('wikia_session_id') : '';
+let cachedItems;
+
+function cacheItems(items) {
+  localStorageConnector.setItem(CACHE_KEY, JSON.stringify({ [CURRENT_SESSION_KEY]: items }));
+}
+
+function get() {
+  if (cachedItems) {
+    return cachedItems;
+  }
+
+  let cachedData;
+
+  try {
+    cachedData = JSON.parse(localStorageConnector.getItem(CACHE_KEY));
+  } catch (e) {
+    /* noop */
+  }
+
+  cachedItems = cachedData ? cachedData[CURRENT_SESSION_KEY] : [];
+
+  return cachedItems;
+}
+
+/**
+ * @param itemId String - created like `wikiId_articleId`
+ */
+function update(itemId) {
+  cacheItems(get().concat(itemId));
+}
+
+function remove(itemCount) {
+  cacheItems(get().slice(itemCount));
+}
+
+function clear() {
+  const items = get();
+
+  if (!items) {
```

Comment:
Shouldn't the condition be reversed here?

Reply:
nah.

eg.
```
sessionId: 123
localStorage: { recommendation_blacklist: { 123: [ ... ] } }
get() => [ ... ]
...
sessionId changes to 234
get() => null
```
so if there are no items related to specific sessionId, we can clear the whole object